pkg/sql/storageparam/tablestorageparam/table_storage_param.go

```diff
@@ -27,25 +27,38 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/storageparam"
 	"github.com/cockroachdb/cockroach/pkg/util/duration"
 	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/cockroachdb/errors"
 )
 
 // Setter observes storage parameters for tables.
 type Setter struct {
-	tableDesc *tabledesc.Mutable
+
+	// TableDesc is the table that is modified by the Setter.
+	TableDesc *tabledesc.Mutable
+
+	// UpdatedRowLevelTTL is kept separate from the RowLevelTTL in TableDesc
+	// in case changes need to be made in schema changer.
+	UpdatedRowLevelTTL *catpb.RowLevelTTL
```

Comment:
Instead of always modifying the TableDesc in `Set()`/`Reset()` and possibly unsetting it in `handleTTLStorageParams()`. It will only be set in `handleTTLStorageParams()` if it is not done in a mutation.